### PR TITLE
Guess right v1

### PR DESCRIPTION
**Pull request by geoffrey112**

### CODE REVIEW

File: src/js/script.js [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+
+  set randomNb(value){
+    this._randomNb = value;
+  }
+
+  get functionality(){
+    return this._functionality;
+  }
+
+  set functionality(value){
```
Comment (wehrli) [inline src/js/script.js]: why do you call it `set functionnality` if you don't touch the variable functionnality ?

File: src/js/script.js [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+    if(this._difficulty === 'easy'){
+      if(this._nbOfLife > 0){
+        if(value === this._randomNb){
+          document.getElementById("answer").innerHTML = "Gagnée!";
+          document.getElementById('selectChance').disabled = true;
+          document.getElementById('difficulty').disabled = true;
```
Comment (wehrli) [inline src/js/script.js]: Those elem which are display could have been in the class `Display`

File: src/js/script.js [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+          this._nbOfLife = 0;
+        }else if(value != this._randomNb && this._nbOfLife === 3){
+          document.getElementById("life1").style.visibility = "hidden";
+          document.getElementById("answer").innerHTML = "Ouch, try again, 2 chance remaining";
+          document.getElementById("difficulty").disabled = true;
+          document.getElementById("selectChance").disabled = true;
```
Comment (wehrli) [inline src/js/script.js]: same here

File: src/js/script.js [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+          --this._nbOfLife;
+        }else if(value !== this._randomNb && this._nbOfLife === 2){
+          document.getElementById("answer").innerHTML = "Ouch, try again.. Last chance";
+          document.getElementById("life2").style.visibility = "hidden";
+          document.getElementById("difficulty").disabled = true;
+          document.getElementById("selectChance").disabled = true;
```
Comment (wehrli) [inline src/js/script.js]: same here

File: src/js/script.js [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+          document.getElementById("answer").innerHTML = "Gagnée!";
+          document.getElementById('selectChance').disabled = true;
+          document.getElementById('difficulty').disabled = true;
+          this._nbOfLife = 0;
+        }else if(value != this._randomNb && this._nbOfLife === 3){
+          document.getElementById("life1").style.visibility = "hidden";
+          document.getElementById("answer").innerHTML = "Ouch, try again, 2 chance remaining";
+          document.getElementById("difficulty").disabled = true;
+          document.getElementById("selectChance").disabled = true;
+          this._randomNb = Math.floor(Math.random()* 11);
```
Comment (wehrli) [inline src/js/script.js]: this could have been a method, avoiding repeating the code

File: src/js/script.js [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+      document.getElementById("difficulty").value = "easy";
+      document.getElementById("life2").style.visibility = "hidden";
+      document.getElementById("life1").style.visibility = "hidden";
+      document.getElementById("life3").style.visibility = "visible";
+      document.getElementById("answer").innerHTML = "";
+      document.getElementById("difficulty").disabled = false;
+      document.getElementById("selectChance").disabled = false;
+      document.getElementById("selectChance").value = 1;
+      document.getElementById("ok").disabled = true;
+      document.getElementById("findNb").value = "";
```
Comment (wehrli) [inline src/js/script.js]: Again this could have been a method from the class `Display`

**Review by wehrli — COMMENTED** [review]
Hey nice introduction into classes. I guess you need to practice a bit more. You did separate the concerned but not enough, you still have some rendering behaviour in the Game class which could have been avoided

File: src/js/script.js (hunk 2 above)
Comment (geoffrey112) [reply src/js/script.js]: Oh, my bad !

**Pull request merged**